tls/s2n_client_cert_verify.c

```diff
@@ -32,14 +32,16 @@ int s2n_client_cert_verify_recv(struct s2n_connection *conn)
     POSIX_ENSURE_REF(hashes);
 
     struct s2n_stuffer *in = &conn->handshake.io;
-    struct s2n_signature_scheme *chosen_sig_scheme = &conn->handshake_params.client_cert_sig_scheme;
 
     if (conn->actual_protocol_version < S2N_TLS12) {
-        POSIX_GUARD(s2n_choose_default_sig_scheme(conn, chosen_sig_scheme, S2N_CLIENT));
+        POSIX_GUARD(s2n_choose_default_sig_scheme(conn,
+                &conn->handshake_params.client_cert_sig_scheme, S2N_CLIENT));
     } else {
         /* Verify the SigScheme picked by the Client was in the preference list we sent (or is the default SigScheme) */
-        POSIX_GUARD(s2n_get_and_validate_negotiated_signature_scheme(conn, in, chosen_sig_scheme));
+        POSIX_GUARD(s2n_get_and_validate_negotiated_signature_scheme(conn, in,
+                &conn->handshake_params.client_cert_sig_scheme));
     }
+    const struct s2n_signature_scheme *chosen_sig_scheme = conn->handshake_params.client_cert_sig_scheme;
```

Comment:
Just making sure: you don't need to ensure ref this (and other similar instances) because it can never be null, right? It's set to the null scheme on initialization, and will never be set to null?

Reply:
Technically yeah it should never be null so we probably don't need to ensure ref, but we still should since we're going to dereference it. I added the checks.